diff --git a/node_modules/whatsapp-web.js/src/util/Injected/Utils.js b/node_modules/whatsapp-web.js/src/util/Injected/Utils.js
index 1bf5704..9787498 100644
--- a/node_modules/whatsapp-web.js/src/util/Injected/Utils.js
+++ b/node_modules/whatsapp-web.js/src/util/Injected/Utils.js
@@ -941,12 +941,12 @@ exports.LoadUtils = () => {
     };
 
     window.WWebJS.getAddParticipantsRpcResult = async (groupMetadata, groupWid, participantWid) => {
-        const participantLidArgs = groupMetadata?.isLidAddressingMode
+        const participantLidArgs = groupMetadata?.isLidAddressingMode && participantWid.server == 'lid'
             ? {
-                phoneNumber: participantWid,
-                lid: window.Store.LidUtils.getCurrentLid(participantWid)
+                phoneNumber: window.Store.LidUtils.getPhoneNumber(participantWid),
+                lid: participantWid
             }
-            : { phoneNumber: participantWid };
+            : { phoneNumber: participantWid.server == 'lid' ? window.Store.LidUtils.getPhoneNumber(participantWid) : participantWid };
 
         const iqTo = window.Store.WidToJid.widToGroupJid(groupWid);
 
